test/dlc_tests/container_tests/bin/testTF2HVD.py

```diff
@@ -48,7 +48,7 @@
 loss = tf.losses.SparseCategoricalCrossentropy()
 
 # Horovod: adjust learning rate based on number of GPUs.
-opt = tf.optimizers.Adam(0.001 * hvd.size())
+opt = tf.optimizers.legacy.Adam(0.001 * hvd.size())
```

Comment:
whats the purpose of this change, is this backward incompatible?

Reply:
With TF2.11, the updates to optimizers cause breaking changes to older optimizers seen in TF2.10 and below. This will not be backwards compatible, so I will look for a way around this issue.

Reply:
Added logic for checking TF versions before assigning optimizer calls